docs/root/intro/arch_overview/load_balancing.rst

```diff
@@ -249,6 +249,28 @@ with regard to percentage relations in the local zone between originating and up
 
 Note that when using multiple priorities, zone aware routing is currently only supported for P=0.
 
+.. _arch_overview_load_balancing_locality_weighted_lb:
+
+Locality weighted load balancing
+--------------------------------
+
+Another approach to determining how to weight assignments across different zones
+and geographical locations is by using explicit weights supplied via EDS in the
+:ref:`LocalityLbEndpoints <envoy_api_msg_endpoint.LocalityLbEndpoints>` message.
+This approach is mutually exclusive with the above zone aware routing, since in
+the case of locality aware LB, we rely on the management server to provide the
+locality weighting, rather than the Envoy-side heuristics used in zone aware
+routing.
+
+Envoy will continue to take into account the health of endpoints in a locality
+and adjust the management server assigned weights based on this. For example,
+if locality X is weighted 60%, locality Y at 20% and locality Z at 20%, but
+only half of locality Z's endpoints are healthy, we would weight X/Y/Z as
+67%/22%/11%.
```

Comment:
It's a bit unclear to me how health comes into weighting. Can you spell out the math for the reader? I think that would help.

Also, per our offline convo with @alyssawilk, how can we potentially have this same algorithm be used for priority failover selection? I think it would be really nice to have the same heuristics/algorithms used (possibly tunable in the future).

Reply:
P.S., I would model the docs after @alyssawilk amazing docs on priority failover: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/load_balancing#priority-levels (and really again related because IMO the algorithm should basically be the same, at least logically, possibly with different knobs).

Reply:
Your call, but I found on the loadbalancing PRs that doing the code and docs in parallel made sense, as I coded things up as a combination of how I wanted them and how Envoy affected our internal LB, and it resulted in much less doc thrashing :-P